### ISSUE TITLE
Title is missing from share

### ISSUE THREAD

**Issue by yaelirub** [Automattic/pocket-casts-android#1281]
### Description

Title is missing when sharing an episode or a podcast. Only URL is visible and when sharing to Buffer, even URL is not visible.
Could this be as a result of "Swipe to Share"?

### Step-by-step reproduction instructions

1. try to share an episode 
2.See only URL, no title

### Screenshots or screen recording

Android             |  iOS
:-------------------------:|:-------------------------:
![](https://github.com/Automattic/pocket-casts-android/assets/1335657/f1b5ed74-718c-4597-9033-63f371b78e4a) |![](https://github.com/Automattic/pocket-casts-android/assets/1335657/33094e84-c551-43f7-9d32-9ed03243cf5c)

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

**Comment by CookieyedCodes** [comment]
@yaelirub not a result of swipe to share it's allways been the case on android but i agree the url/uuid should be translated for the share sheet even if the user is sharing the url (and maby a setting to paste title in the open file in popup, I kind of don't like I have to rename episodes every time when I'm clipping audio 😅)

**Comment by joenjogu** [comment]
@CookieyedCodes Android added support for rich share previews in Android 10 described [here](https://developer.android.com/training/sharing/send#adding-rich-content-previews)

I've done a bare minimum impl and can display the episode/podcast title in the share sheet as you can see below:
![pocketcasts_share_sheet](https://github.com/Automattic/pocket-casts-android/assets/20142549/2fcc6e40-e5a9-49bf-b08c-984e22fc96c7)

I can pick this task up. The only issue is I might need a bit of guidance from @mchowning or @ashiagr on how to display the podcast thumbnail in the share sheet.
It requires a fileprovider URI set as clipData to the share Intent. My first thought was to download the bitmap using Coil then cache the image and retrieve a URI for it though this seems a bit expensive and there might already be an impl on the app to cache podcast thumbnails.

**Comment by mchowning** [comment]
> My first thought was to download the bitmap using Coil then cache the image and retrieve a URI for it though this seems a bit expensive and there might already be an impl on the app to cache podcast thumbnails.

It would be nice if we could avoid saving the file to disk, but if we need to do that, your approach seems reasonable to me. I might be mistaken, but I don't think we do any caching of images in the app outside of Coil, and I believe Coil relies on okhttp, which means it will only cache images based on the HTTP headers (meaning some images might not get cached). The `CoilManager` and `PodcastImageLoader` might be worth taking a look at to see how we're handling images already in the app.

If we do save the files to disk, let's just make sure we clean them up at some point when we don't need them anymore. I wouldn't want these files just sticking around forever.

**Comment by joenjogu** [comment]
@mchowning from a quick search I can see the app caches user-subscribed podcast images to disk using Coil and refreshes the cache with a worker (there's a way to clear as well).
My first thought was to see if Coil can allow us to cache and retrieve the displayed thumbnail as a URI, but this is looking increasingly unachievable.
You can assign me the task and I can take a deep dive into it and if all else fails, I'll use the download, cache, URI implementation I mentioned above.

**Comment by mchowning** [comment]
> My first thought was to see if Coil can allow us to cache and retrieve the displayed thumbnail as a URI, but this is looking increasingly unachievable

Yeah, I didn't see any obvious way to accomplish that either in my quick search.

> You can assign me the task and I can take a deep dive into it and if all else fails, I'll use the download, cache, URI implementation I mentioned above.

Sounds good. Thanks for tackling this! 🙇
